src/auslib/db.py

```diff
@@ -2944,6 +2945,73 @@ def delete(self, where, changed_by=None, old_data_version=None, transaction=None
         super(EmergencyShutoffs, self).delete(changed_by=changed_by, where=where, old_data_version=old_data_version, transaction=transaction, dryrun=dryrun)
 
 
+class PinnableReleasesTable(AUSTable):
+    def __init__(self, db, metadata, dialect):
+        self.table = Table(
+            "pinnable_releases",
+            metadata,
+            Column("product", String(15), nullable=False, primary_key=True),
+            Column("version", String(75), nullable=False, primary_key=True),
+            Column("channel", String(75), nullable=False, primary_key=True),
+            Column("mapping", String(100), nullable=False),
+        )
+        AUSTable.__init__(self, db, dialect, scheduled_changes=True, scheduled_changes_kwargs={"conditions": ["time"]}, historyClass=HistoryTable)
+
+    def getPotentialRequiredSignoffs(self, affected_rows, transaction=None):
+        # Implementing this is required to schedule changes to this table
+        return None
+
+    def insert(self, changed_by, transaction=None, dryrun=False, **columns):
+        # XXX check the release exists?
+        # XXX signoffs?
```

Comment:
What's the plan for resolving these?

Reply:
Perhaps @jcristau can speak more to these comments, since he wrote them. It's not clear to me that we need to check if the release exists. If we do, it seems reasonable enough to either make the mapping a foreign key or to simply add a check that that mapping exists.

I'm going to completely defer to @jcristau on the signoffs issue. I believe that we discussed this during the creation of this patch, and IIRC, it was still an open question as to whether signoffs are necessary for this. And I don't feel like I have a deep enough understanding of the issue to have a particularly good opinion about it.

Reply:
At this point I'm reasonably confident we don't need/want signoffs for pins.  They'd add extra friction for each release, require admin UI work, and I don't think there's that much value.